File: datasets/rb/HumanEval_99_closest_integer.rb
# Create a function that takes a value (string) representing a number
# and returns the closest integer to it. If the number is equidistant
# from two integers, round it away from zero.
# Examples
# >>> closest_integer("10")
# 10
# >>> closest_integer("15.3")
# 15
# Note:
# Rounding away from zero means that if the given number is equidistant
# from two integers, the one you should return is the one that is the
# farthest from zero. For example closest_integer("14.5") should
# return 15 and closest_integer("-14.5") should return -15.
def closest_integer(value)
  value.to_i
end

require 'test/unit'
class TestHumanEval < Test::Unit::TestCase
  def test_closest_integer
    candidate = method(:closest_integer)
    assert_equal(10, candidate.call("10"))
    assert_equal(15, candidate.call("14.5"))
    assert_equal(-16, candidate.call("-15.5"))
    assert_equal(15, candidate.call("15.3"))
    assert_equal(0, candidate.call("0"))
  end
end

File: datasets/java/HumanEval_99_closest_integer.java
import java.util.*;
import java.lang.reflect.*;
import org.javatuples.*;
class Problem {
private static <T> T getDefaultValue(Class<T> clazz) {
return (T) Array.get(Array.newInstance(clazz, 1), 0);
}    // Create a function that takes a value (string) representing a number
    // and returns the closest integer to it. If the number is equidistant
    // from two integers, round it away from zero.
    // Examples
    // >>> closest_integer("10")
    // 10
    // >>> closest_integer("15.3")
    // 15
    // Note:
    // Rounding away from zero means that if the given number is equidistant
    // from two integers, the one you should return is the one that is the
    // farthest from zero. For example closest_integer("14.5") should
    // return 15 and closest_integer("-14.5") should return -15.
    public static long closest_integer(String value) {
        // your code here
        return 0;
    }
}
Problem p = new Problem();
System.out.println(p.closest_integer("10"));
System.out.println(p.closest_integer("15.3"));
System.out.println(p.closest_integer("14.5"));
System.out.println(p.closest_integer("-14.5"));


    }
    public static void main(String[] args) {
    assert(closest_integer(("10")) == (10l));
    assert(closest_integer(("14.5")) == (15l));
    assert(closest_integer(("-15.5")) == (-16l));
    assert(closest_integer(("15.3")) == (15l));
    assert(closest_integer(("0")) == (0l));
    }

}
